forest.hs

```diff
@@ -1,40 +1,41 @@
-{-# LANGUAGE BangPatterns #-}
-import Data.List
+module Main where
+
+import qualified Data.Set as S
 import System.Environment
 
 data Forest = Forest
   { goats :: !Int
   , wolves :: !Int
   , lions :: !Int
```

Comment:
oh, quick question; you've removed bang patterns, does these bangs still mean anything without the extension?